base/multidimensional.jl

```diff
@@ -879,6 +879,28 @@ _maybe_linear_logical_index(::IndexLinear, A, i) = LogicalIndex{Int}(i)
 uncolon(::Tuple{}) = Slice(OneTo(1))
 uncolon(inds::Tuple) = Slice(inds[1])
 
+"""
+    _checked_iterate(iter[, state])
+
+Internal function used to eliminate the dead branch in `iterate`.
+Fallback to `iterate` by default, but optimized for indices type in `Base`.
+"""
+@propagate_inbounds _checked_iterate(iter) = iterate(iter)
+@propagate_inbounds _checked_iterate(iter, state) = iterate(iter, state)
```

Comment:
```suggestion
@propagate_inbounds _checked_iterate(iter) = something(iterate(iter))
@propagate_inbounds _checked_iterate(iter, state) = something(iterate(iter, state))
```

Reply:
We can't do this as the fallback `iterate` should return `nothing` for the end iteration.

Reply:
Ah, right. It is more sketchy that it is actually a `_postchecked_iterate`, but that is currently the usages